### PR TITLE
Rename replay core classes

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

> **Naming**: The naming ReplayHistoryLocationEngine suggests that it can only replay history traces, but with the modular design using the mappers, it can also play DirectionsRoute s. Therefore, the name ReplayLocationEngine would fit better here. What do you think?

With the introduction of replay route, the ReplayHistoryLocationEngine and ReplayHistoryPlayer has created confusion.

Renaming these classes to
`ReplayLocationEngine`: capable of replaying history, routes, and more
`MapboxReplayer`: capable of replaying history, routes, and more

- [ ] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventSimulator.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import kotlinx.coroutines.launch
  * This class keeps track of a forward playing replay. As time moves forward, it captures
  * all events from [ReplayEvents] that happened, and provides them in a [ReplayEvents]
  *
- * @param replayEvents events needed to be replayed by [ReplayHistoryPlayer]
+ * @param replayEvents events needed to be replayed by [MapboxEventPlayer]
  * @param logger interface for logging any events
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventSimulator.kt:18]:
```suggestion
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventSimulator.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ import kotlinx.coroutines.launch
  * This class keeps track of a forward playing replay. As time moves forward, it captures
  * all events from [ReplayEvents] that happened, and provides them in a [ReplayEvents]
  *
- * @param replayEvents events needed to be replayed by [ReplayHistoryPlayer]
+ * @param replayEvents events needed to be replayed by [MapboxEventPlayer]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventSimulator.kt:17]:
```suggestion
 * @param replayEvents events needed to be replayed by [MapboxReplayer]
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventsObserver.kt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package com.mapbox.navigation.core.replay.history
 
 /**
- * Used to observe events replayed by the [ReplayHistoryPlayer]
+ * Used to observe events replayed by the [MapboxEventPlayer]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEventsObserver.kt:4]:
```suggestion
 * Used to observe events replayed by the [MapboxReplayer]
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt [hunk 4]
```diff
@@ -12,7 +12,7 @@ interface CustomEventMapper {
 
     /**
      * Override to map your own custom events from history files,
-     * into [ReplayEventBase] for the [ReplayHistoryPlayer]
+     * into [ReplayEventBase] for the [MapboxEventPlayer]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt:15]:
```suggestion
     * into [ReplayEventBase] for the [MapboxReplayer]
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt [hunk 5]
```diff
@@ -32,15 +32,15 @@ class ReplayHistoryMapper @JvmOverloads constructor(
 ) {
 
     /**
-     * Given raw json string return [ReplayEvents] that can be given to a [ReplayHistoryPlayer]
+     * Given raw json string return [ReplayEvents] that can be given to a [MapboxEventPlayer]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt:35]:
```suggestion
     * Given raw json string return [ReplayEvents] that can be given to a [MapboxReplayer]
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt [hunk 6]
```diff
@@ -32,15 +32,15 @@ class ReplayHistoryMapper @JvmOverloads constructor(
...
      */
     fun mapToReplayEvents(historyData: String): List<ReplayEventBase> {
         val exampleHistoryData = gson.fromJson(historyData, ReplayHistoryDTO::class.java)
         return mapToReplayEvents(exampleHistoryData)
     }
 
     /**
-     * Given [ReplayHistoryDTO] return [ReplayEvents] that can be given to a [ReplayHistoryPlayer]
+     * Given [ReplayHistoryDTO] return [ReplayEvents] that can be given to a [MapboxEventPlayer]
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt:43]:
```suggestion
     * Given [ReplayHistoryDTO] return [ReplayEvents] that can be given to a [MapboxReplayer]
```

**Comment by kmadsen** [comment]
cc: @langsmith more doc updates

**Pull request merged**